github/api.go

```diff
@@ -102,7 +102,12 @@ func (a *Client) decode(url string, dest interface{}) error {
 		return errors.WithStack(err)
 	}
 	defer resp.Body.Close()
-	err = json.NewDecoder(resp.Body).Decode(dest)
+	if resp.StatusCode < 200 || resp.StatusCode > 299 {
+		return errors.Errorf("GitHub API request failed with %s", resp.Status)
+	}
+	dec := json.NewDecoder(resp.Body)
+	dec.DisallowUnknownFields()
```

Comment:
Why do we want this? Feels like unnecessary fragility

Reply:
That...is a good point, it was motivated by the fact that the error response from GitHub is a JSON blob that this happily accepted...